Sources/NetworkProtection/Networking/NetworkProtectionClient.swift

```diff
@@ -171,9 +171,14 @@ final class NetworkProtectionBackendClient: NetworkProtectionClient {
     init(environment: VPNSettings.SelectedEnvironment = .default, isSubscriptionEnabled: Bool) {
         self.isSubscriptionEnabled = isSubscriptionEnabled
 
-        // todo - https://app.asana.com/0/0/1206470585910129/f
         if isSubscriptionEnabled {
-            self.endpointURL = URL(string: "https://staging1.netp.duckduckgo.com")!
+            // todo - https://app.asana.com/0/0/1206811466624632/f
+            let usesSubscriptionStaging = true
+            if usesSubscriptionStaging {
+                self.endpointURL = VPNSettings.SelectedEnvironment.staging.endpointURL
+            } else {
+                self.endpointURL = VPNSettings.SelectedEnvironment.production.endpointURL
+            }
```

Comment:
When subscription is enabled, the NetP environment has to correspond to its subscription counterpart. Passing subscription endpoint all the way from outside BSK would require a lot of changes, so in the interest of time, I just hard code the decision here. Once subscription endpoint is set to production, I'd flip this variable to `false`.

Reply:
Since we have to match the subscription counterpart, any reason not to use `SubscriptionPurchaseEnvironment.currentServiceEnvironment` like is done [here](https://github.com/duckduckgo/BrowserServicesKit/blob/ac2127a26f75b2aa293f6036bcdd2bc241d09819/Sources/Subscription/URL%2BSubscription.swift#L24C16-L24C73)? That way we match the current environment for free.

Reply:
Practical reason. We have to `import Subscription` inside BSK and there's no way to do that within an `#if SUBSCRIPTION` block. It'll be doable once we no longer need to check for that `SUBSCRIPTION` flag.